mendes/src/cookies.rs

```
@@ -145,25 +145,27 @@ where
 }
 
 fn extract<T: CookieData>(key: &Key, headers: &HeaderMap) -> Option<T> {
-    let cookies = headers.get("cookie")?;
-    let cookies = str::from_utf8(cookies.as_ref()).ok()?;
+    let cookies = headers.get_all("cookie");
     let name = T::NAME;
-    for cookie in cookies.split(';') {
-        let cookie = cookie.trim_start();
-        if cookie.len() < (name.len() + 1 + NONCE_LEN + TAG_LEN)
-            || !cookie.starts_with(name)
-            || cookie.as_bytes()[name.len()] != b'='
-        {
-            continue;
-        }
+    for cookie_header in cookies {
```

I think I'd prefer to call `cookie_header` `value` instead, and we can probably just inline the call to `headers.get_all()` in the loop header?

If we fail to decode the UTF-8, we should continue rather than returning `None`:

```rust
let Ok(value) = str::from_utf8(value.as_ref()) {
    continue;
};
```